lib/plugins/rotate_asg_instances.rb

```diff
@@ -7,11 +7,41 @@ module Moonshot
   module Plugins
     # Rotate ASG instances after update.
     class RotateAsgInstances
-      def post_update(resources)
-        asg = ASG.new(resources)
+      include ::Moonshot::DoctorHelper
+
+      def initialize(ssh_user = ENV['MOONSHOT_SSH_USER'])
+        @ssh_user = ssh_user
+      end
+
+      def doctor(resources)
+        @resources = resources
+        doctor_check_ssh
+      end
+
+      def pre_update(resources)
+        @resources = resources
+        asg.try_to_ssh(raise_exception: true)
+      end
+
+      def post_update(_resources)
         asg.rotate_asg_instances
```

Comment:
Looks like the asg method needs the resources, can we pass it explicit as a parameter instead setting a instance variable, I think this way we avoid bugs like that and if we still forget the error will be clear saying we need to send resources to the method.

Reply:
Made it a parameter and removed the instance variable implementation.